client/src/components/EditMyAvailability.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import { Form, Modal } from "react-bootstrap";
+import Button from "react-bootstrap/Button";
+function EditMyAvailability(props) {
+  const [edit, setEdit] = useState(false);
+  const [trainees_id, setTrainees_id] = useState(localStorage.getItem("id"));
+  const [date, setDate] = React.useState(new Date());
+  const [topic, setTopic] = React.useState("");
+  const handleUpdate = (e) => {
+    e.preventDefault();
+    const formattedDate = `${date.getFullYear()}-${(
+      "0" +
+      (date.getMonth() + 1)
+    ).slice(-2)}-${("0" + date.getDate()).slice(-2)}`;
+    const available = {
+      availability_date: formattedDate,
+      topic: topic,
+      trainees_id: trainees_id,
+    };
+    fetch(`https://study-buddies.onrender.com/availability/${props.id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: localStorage.getItem("token"),
+      },
+      body: JSON.stringify(available),
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.error) {
+          alert(data.error);
+        } else {
+          alert("You have updated this available date!");
```

Comment:
could we please not use window alert? Could this be a banner component that displays this message on top of the page (for example on timeout something like 5 seconds?)

Reply:
I created errorMessage useState for this. I can't see what the page is doing at the moment but I hope that solves the problem.